solr/modules/cross-dc/src/java/org/apache/solr/crossdc/update/processor/MirroringUpdateProcessor.java

```diff
@@ -233,7 +223,9 @@ public void processDelete(final DeleteUpdateCommand cmd) throws IOException {
         boolean done = false;
         while (!done) {
           q.set(CursorMarkParams.CURSOR_MARK_PARAM, cursorMark);
-          QueryResponse rsp = client.query(collection, q);
+          var baseUrl = cmd.getReq().getCore().getCoreContainer().getZkController().getBaseUrl();
+          var client = cmd.getReq().getCoreContainer().getDefaultHttpSolrClient();
+          QueryResponse rsp = client.requestWithBaseUrl(baseUrl, c -> c.query(collection, q));
```

Comment:
@HoustonPutman  wouldn't it be cool if this could talk directly without HTTP?  I suppose EmbeddedSolrServer might not be appropriate since it doesn't handle collection arguments.  But the collection in this case appears to be the same collection as the present core, thus we could address this to the same core?